### PR TITLE
Update recipe rerun results and assets

### PR DESCRIPTION
**Pull request by derek-tml**
## Summary
- refresh recipe README results for chat SL, code RL, math RL, rubric, SDFT, and true-thinking-score
- add rubric reward plot assets and rubric grader parsing fixes
- add the Harbor eval helper and update Harbor recipe docs/scripts

### CODE REVIEW

**Review by ZitongYang — APPROVED** [review]
`harbor_rl` looks good to me!

File: tinker_cookbook/recipes/chat_sl/README.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ python -m tinker_cookbook.recipes.chat_sl.train \
     wandb_project=cookbook_sl
 ```
 
-After 1740 steps of training, `test/nll` decreases to 0.50.
+After 7000 steps of training, `test/nll` decreases to 0.641.
```
Comment (nealwu) [inline tinker_cookbook/recipes/chat_sl/README.md]: 👀 why did we increase training steps by so much? (how long does this take to run?)

File: tinker_cookbook/recipes/code_rl/README.md [hunk 2]
```diff
@@ -49,22 +49,23 @@ Optional environment variables for Modal:
...
-    model_name="Qwen/Qwen3-4B-Instruct-2507" \
+    model_name="Qwen/Qwen3.5-4B" \
     group_size=8 groups_per_batch=128 \
     learning_rate=4e-5 \
     lora_rank=32 \
     max_tokens=24576
 ```
 
-After 100 steps of training, you can expect the following performance on **LiveCodeBench v6 (2025.02–2025.05)**:
+After 190 steps of training, you can expect the following performance on **LiveCodeBench v6 (2025.02–2025.05)**:
```
Comment (nealwu) [inline tinker_cookbook/recipes/code_rl/README.md:63]: similar question here

File: tinker_cookbook/recipes/code_rl/README.md [hunk 3]
```diff
@@ -49,22 +49,23 @@ Optional environment variables for Modal:
...
 
 | Model | Pass@1 | Pass@8 |
 |-------|--------|--------|
-| Qwen3-4B-Instruct-2507 (before training) | 33.8% | 44.3% |
-| Qwen3-4B-Instruct-2507 (after 100 steps) | 42.7% | 55.0% |
+| Qwen3.5-4B (before training) | 23.6% | 34.3% |
+| Qwen3.5-4B (after 190 steps) | 52.4% | 76.0% |
```
Comment (nealwu) [inline tinker_cookbook/recipes/code_rl/README.md:68]: interesting, so it was actually worse before training but then even better afterward

File: tinker_cookbook/recipes/harbor_rl/README.md [hunk 4]
```diff
@@ -111,22 +118,27 @@ uvx harbor datasets download swebench-verified@1.0 -o ~/.cache/harbor/tasks/sweb
...
 
 | Benchmark | Total | PASS | FAIL | ERROR | Pass Rate |
 |-----------|-------|------|------|-------|-----------|
-| SWE-Bench Verified 1.0 | 500 | 46 (9.2%) | 52 (10.4%) | 402 (80.4%) | 9.2% |
-| Terminal-Bench 2.0 | 89 | 18 (20.2%) | 36 (40.4%) | 35 (39.3%) | 20.2% |
+| SWE-Bench Verified 1.0 | 500 | 145 (29.0%) | 52 (10.4%) | 303 (60.6%) | 29.0% |
+| Terminal-Bench 2.0 | 89 | 14 (15.7%) | 31 (34.8%) | 44 (49.4%) | 15.7% |
 
-**Config**: `max_turns=200, max_tokens=8192, temperature=0.1, sandbox_timeout=3600s`
+**Config**: `max_turns=200, max_tokens=8192, temperature=1.0, sandbox_timeout=3600s`
```
Comment (nealwu) [inline tinker_cookbook/recipes/harbor_rl/README.md:141]: why the major temperature change?

File: tinker_cookbook/recipes/math_rl/README.md [hunk 5]
```diff
@@ -13,65 +13,39 @@ uv pip install 'tinker-cookbook[math-rl] @ git+https://github.com/thinking-machi
...
-
-- 3+5+13=21
-
-- 3+7+11=21
-
-- 5+7+11=23
-
-So the best possible is sum1=20 and sum2=21, product 420. So the maximum sum is 420. \boxed{420}</think>
+The point (0,3): x=0, y=3. r = sqrt(0^2+3^2)=3. \theta = arctan(y/x) but x=0, y>0 so \theta=\pi/2. So (3, \pi/2). Format: \boxed{(3,\frac{\pi}{2})}.
+</think>
```
Comment (nealwu) [inline tinker_cookbook/recipes/math_rl/README.md:37]: where did we get these new examples from?

File: tinker_cookbook/recipes/rubric/env.py [hunk 6]
```diff
@@ -229,11 +228,14 @@ class RubricGradedDatasetBuilder(RLDatasetBuilder):
...
-    grader_llm_name: str = "Qwen/Qwen3-30B-A3B-Instruct-2507"
+    grader_llm_name: str = "Qwen/Qwen3.6-35B-A3B"
+    grader_renderer_name: str | None = None
 
     def _get_grader_llm(self) -> MessageCompleter:
         tokenizer = get_tokenizer(self.grader_llm_name)
-        renderer_name = model_info.get_recommended_renderer_name(self.grader_llm_name)
+        renderer_name = self.grader_renderer_name or model_info.get_recommended_renderer_name(
+            self.grader_llm_name
+        )
```
Comment (nealwu) [inline tinker_cookbook/recipes/rubric/env.py:238]: as a heads up this returns the thinking renderer by default. not saying that's good or bad, just double checking that's what you wanted

**Review by nealwu — APPROVED** [review]

File: tinker_cookbook/recipes/rubric/env.py (hunk 6 above)
Comment (nealwu) [reply tinker_cookbook/recipes/rubric/env.py:238]: based on https://github.com/thinking-machines-lab/tinker-cookbook/blob/b9a494708cae396460d7ed5f842fb6eba723ea7e/tinker_cookbook/model_info.py#L24 (first one is the default)

File: tinker_cookbook/recipes/harbor_rl/README.md (hunk 4 above)
Comment (derek-tml) [reply tinker_cookbook/recipes/harbor_rl/README.md:141]: the reason is because training happens at temperature=1.0, so i changed the eval to match. i believe it should always have been this.

File: tinker_cookbook/recipes/math_rl/README.md (hunk 5 above)
Comment (derek-tml) [reply tinker_cookbook/recipes/math_rl/README.md:37]: this polar coordinate is item 0 of the MATH-500 dataset, just pulled out first one for the new renderer

File: tinker_cookbook/recipes/rubric/env.py (hunk 6 above)
Comment (derek-tml) [reply tinker_cookbook/recipes/rubric/env.py:238]: yeah this is fine, i explicitly pass in self.grader_renderer_name = qwen3_5_disable_thinking as default anyways.

File: tinker_cookbook/recipes/code_rl/README.md (hunk 2 above)
Comment (derek-tml) [reply tinker_cookbook/recipes/code_rl/README.md:63]: 190 is also by default if you use batch size 128. maybe they used 256 before, or just used an intermediate checkpt. either way i think not a huge deal for this one.

File: tinker_cookbook/recipes/chat_sl/README.md (hunk 1 above)
Comment (derek-tml) [reply tinker_cookbook/recipes/chat_sl/README.md]: the script goes to 7000 by default if we use batch_size=128. if they used batch_size=512 before it would be 1740. let me rerun it with that batch size since its quite off
